pkg/rbd/controllerserver.go

```diff
@@ -0,0 +1,103 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package rbd
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/pborman/uuid"
+	"golang.org/x/net/context"
+
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/container-storage-interface/spec/lib/go/csi"
+	"github.com/kubernetes-csi/drivers/pkg/csi-common"
+)
+
+type controllerServer struct {
+	*csicommon.DefaultControllerServer
+	clientSet *kubernetes.Clientset
+}
+
+func GetVersionString(ver *csi.Version) string {
+	return fmt.Sprintf("%d.%d.%d", ver.Major, ver.Minor, ver.Patch)
+}
+
+func (cs *controllerServer) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequest) (*csi.CreateVolumeResponse, error) {
+	fmt.Printf("><SB> Create Req: %+v timestamp: %s\n", req, time.Now().String())
+	if err := cs.Driver.ValidateControllerServiceRequest(req.Version, csi.ControllerServiceCapability_RPC_CREATE_DELETE_VOLUME); err != nil {
+		glog.V(3).Infof("invalid create volume req: %v", req)
+		return nil, err
+	}
+	// Volume Name
+	volName := req.GetName()
+	if len(volName) == 0 {
+		volName = uuid.NewUUID().String()
```

Comment:
make it more informative like "rbd-csi-" + uuid

Reply:
should I not match to normal naming schema? like: kubernetes-dynamic-pvc- + uuid

Reply:
it is more a csi volume than k8s

Reply:
in this case, should we overwrite volName for ALL cases and enforce csi-rbd-+uuid schema?